src/Workspaces/Core/Portable/Shared/Extensions/SyntaxGeneratorExtensions.cs

```diff
@@ -84,16 +84,20 @@ internal static partial class SyntaxGeneratorExtensions
                 // call the default constructor.
 
                 return containingTypeOpt.GetMembers()
-                    .OfType<IFieldSymbol>()
-                    .Where(field => !field.IsStatic)
-                    .Select(field => field.AssociatedSymbol ?? field)
+                    .Where(member => GetField(member) is { IsStatic: false })
                     .Except(parameterToExistingFieldMap.Values)
                     .Any();
             }
 
             return false;
         }
 
+        private static IFieldSymbol GetField(ISymbol member)
+        {
+            // TODO: event backing fields
+            return member as IFieldSymbol ?? (member as IPropertySymbol)?.AssociatedField;
```

Comment:
It looks like there's a test hole here as well that would require IEventSymbol.AssociatedField to be exposed.